app/controllers/works_controller.rb

```diff
@@ -62,6 +62,9 @@ def index
       end
 
       tag = @fandom || @tag
+
+      return raise_not_found unless tag
```

Comment:
It'd be preferred to raise RecordNotFound by using `Fandom.find_by!` in line 61. That will be converted to a 404 error in production and is a common pattern in this codebase.

Reply:
@Bilka2 I fixed it by using `find` instead of `find_by!` since `find` will also raise `RecordNotFound` in the same usecase. Does that work for you?